/**:
  ros__parameters:
    # -- System --
    cdob_ctrl_period: 0.033     # sampling period.

    # -- Q-filters -- if the order is 2 or more, we use damping factors/
    qfilter_lateral_error_cdob_order: 1  # must be ==1 as the state observer needed to be redesigned.
    qfilter_lateral_error_cdob_freq: 10.  #qfilter frequency in Hz for lateral error
    qfilter_lateral_cdob_damping: 2.

    qfilter_lateral_dob_order: 4     # disturbance observer qfilter order.
    qfilter_lateral_dob_freq: 10.    # qfilter frequency in Hz for lateral error
    qfilter_lateral_dob_damping: 5. # damping for the dob filter.

    qfilter_longitudinal_error_order: 2  # longitudinal control qfilter.
    qfilter_longitudinal_error_freq: 5.  #qfilter frequency in Hz for heading error

    # -- Vehicle Model Selection -- Currently we have only one.
    steering_time_constant_: 0.27 # first order steering model time-constant  delta_dot = -1/tau *delta + 1/tau * u
    velocity_time_constant_: 0.1 # first order velocity model time-constant
    acc_time_constant_: 0.1 # first order velocity model time-constant

    # Observer parameters for the LPV system
    Xn1: [ -0.00021857036458138604, 0.0012743981744613214, 0.0010116991474823029, -0.0005396400151106543, 0.0012743981744613214, 0.0008696563609508597, 0.0012347211965813315, -0.0003762786347182012, 0.0010116991474823029, 0.0012347211965813315, 0.001429345159079951, -0.0005315516953973381, -0.0005396400151106543, -0.0003762786347182012, -0.0005315516953973381, 0.00022176876895209534 ]
    Xn2: [ 0.05587885287670698, 0.026482105597695028, 0.012018598116811434, -0.010527991528207215, 0.026482105597695028, 0.0029239322204034886, -0.013311933469663672, 0.0022672051730770575, 0.012018598116811434, -0.013311933469663672, -0.03029607744226035, 0.010033305696359473, -0.010527991528207215, 0.0022672051730770575, 0.010033305696359473, -0.002976227232966234 ]
    Xn3: [ 0.00453151573798223, 0.0006968944615271239, 0.002781784301004176, -0.00042325591917412655, 0.0006968944615271239, -9.22732897647629e-06, 0.0008684297164887417, 1.6860154337742515e-05, 0.002781784301004176, 0.0008684297164887417, 0.0022609358542814784, -0.00043458358780589846, -0.00042325591917412655, 1.6860154337742515e-05, -0.00043458358780589846, 6.919806660756771e-05 ]
    Xn4: [ 0.003134496461587088, -8.575012170098293e-05, 0.00026050009706614347, -8.113570462346322e-05, -8.575012170098293e-05, 4.7858453566989056e-05, -0.0007197212538370316, -6.004026243412433e-05, 0.00026050009706614347, -0.0007197212538370316, -0.0011869137558765035, 0.0002710944962662654, -8.113570462346322e-05, -6.004026243412433e-05, 0.0002710944962662654, -4.644577122927404e-05 ]
    Xn5: [ 4.323689469103361, 1.5504330049363375e-05, 0.14675149014757785, -0.04745073708526461, 1.5504330049363375e-05, 4.39378225035224, 0.1445633957288808, -0.06787216374305217, 0.14675149014757785, 0.1445633957288808, 4.974264913281977, 0.3815979653355504, -0.04745073708526461, -0.06787216374305217, 0.3815979653355504, 1.1725657160900098 ]

    Yn1: [ -0.001731741125853509, 0.030015586353956417, -0.03490048392961781, 0.0265250962706142, -0.11225884397654684, 0.19192417318120916, -0.028700528574771022, 0.1347973274956037, -0.18755133409607333, 0.003574993193261738, -0.00785273647213266, 0.02125461374934384 ]
    Yn2: [ -0.059601494189393124, 0.05004279319498974, -0.004038063031340282, -0.006982731472877617, 0.06871746803002121, 0.058077322793211664, 0.01285507199847829, 0.05628342699244414, -0.15707108928136473, 0.00969763550358059, -0.04625216847364646, 0.022124896805937863 ]
    Yn3: [ 0.003659488208952095, 0.027370550197080856, -0.0009926047802203196, -0.008545450722424849, 0.025292422691001536, -0.011900203343712086, 0.00020406375641860924, -0.00576943951655678, 0.05367358860703643, 0.001247687130879322, 0.0014232870450832029, -0.006242086141521845 ]
    Yn4: [ 0.000771807154181982, -0.00717474834480393, 0.06995442125478152, -0.0017353987463640784, 0.11311622271321777, -0.3314394555656999, 0.017789528246800718, -0.22125275834896316, 0.3665171253225766, -0.0038080626536448063, 0.01589292156467347, -0.03533972237073247 ]
    Yn5: [ -2.8847085549473217, -0.7197633791172355, 0.3208881319081925, -0.7197525925625098, -1.7340492868029578, -0.1859060573556654, 0.2651323943106597, -0.1561745074109684, -2.606880263889149, -0.063720879831262, 0.02370288007046134, 0.44253869973252163 ]





